Fix click behavior routing to a different dashboard with a non-default target tab

JesseSDevaney opened the pull request:
- Closes: https://github.com/metabase/metabase/issues/39320

---
### How to Manually Verify
1. Create two dashboards
  - (1) Add a card configured with click behavior routing to the second dashboard with the 2nd tab being the target tab
    ![image](https://github.com/metabase/metabase/assets/22608765/1d7468a2-3661-4f02-9f96-8753f0ae9e43)
  - (2) Create a two tabbed dashboard with a filter applied to a card on each tab
    ![image](https://github.com/metabase/metabase/assets/22608765/0220a7b1-05ef-4523-9ac0-54722dce5b0f)
    
2. Test click behavior on the first dashboard and verify that it routes you to the correct dashboard and correct tab
  ![image](https://github.com/metabase/metabase/assets/22608765/dd49bf33-cf25-4adf-bdc6-f8ac0605f4d9)
  ![image](https://github.com/metabase/metabase/assets/22608765/0f966d6a-1425-4aae-815e-733fe3ec4b1e)
4. Test multiple times: try refreshing on the first dashboard and testing it again. Verify that it routes you to the correct tab on every attempt.

JesseSDevaney committed:
remove unnecessary parameter value setting

- it already occurs when pushing the new location

kulyk commented on frontend/src/metabase/visualizations/lib/action.js:
```diff
@@ -20,7 +19,6 @@ export function performAction(action, { dispatch, onChangeCardAndRun }) {
         openInSameOrigin: location => {
           dispatch(push(location));
-          dispatch(setParameterValuesFromQueryParams(location.query));
         },
```
Doesn't it conflict with #24762? (that PR added this line)
Not sure how its related to tabs

JesseSDevaney replied:
Yes it does, but only when routing to questions. I did not notice this until looking at the failing tests and when working on another issue. Removing that line does fix the race condition of pushing a new dashboard + tab location when performing a click action. Somehow it is competing for setting the URL when this occurs. 

I am investigating alternative fixes because it solves one bug but causes another.

JesseSDevaney committed:
update tests to reflect changes

JesseSDevaney committed:
update type to reflect usage

JesseSDevaney committed:
setup two way sync for parameter values

JesseSDevaney committed:
Merge branch 'master' into 39320/fix-click-behavior-dashboard-race-condition

JesseSDevaney committed:
fix url query param parsing

- it is keeping the question and then when creating a new query string, it adds another (being recognized as %3F = ?)

JesseSDevaney committed:
fix NaN behavior when normally setting values does not result in NaNs

JesseSDevaney committed:
Merge branch 'master' into 39320/fix-click-behavior-dashboard-race-condition

JesseSDevaney commented:
The two-way sync that would solve:
- https://github.com/metabase/metabase/issues/35126
- https://github.com/metabase/metabase/issues/39320
appears to be at odds with a lot of different behaviors (buggy behaviors).

Unsure if we want to continue fixing this in this moment due to the effort and complexity. I pushed up some changes to check to see how many tests were fixed. If there are still a lot failing, I will update the issue with my findings and come back around to this after tackling easier issues.

JesseSDevaney commented on frontend/src/metabase/query_builder/utils.ts:
If this PR ends up being closed, we should still open up a separate PR for this fix in particular
